src/collector/ActionBatchingCollector.js

```diff
@@ -0,0 +1,30 @@
+
+const localStorageActionsKey = 'eggplantRciActions';
+export default class ActionBatchingCollector {
+
+  constructor(localStorage) {
+    this.localStorage = localStorage;
+  }
+
+  /**
+       * @param  {Event} event
+       * @param  {Context} context
+       */
+  async prepare (event, context) {
+    /** Pull out the actions from local storage */
+    const actions = this.localStorage.getItem(localStorageActionsKey);
+
+    /** Append the new action */
+    actions.push(event.eventInfo5[0]);
+    if (context.scope === 'state') {
+
+      /** Stringify actions and attached it to the state beacon */
+      event.eventInfo5 = JSON.stringify(actions);
```

Comment:
@robertdumitrescu INFO - We need to be careful about string length as anything above 1024 chars will be rejected.  This is much lower than I had originally thought...  https://docs.real-user-data.eggplant.cloud/open-api/index.html